### PR TITLE
Comments and minor improvements for utils.ts

### PR DESCRIPTION
**Pull request by vnnkl**
Added some comments to the function to make them more consistent with the others

The `isObj` function in the `utils.ts` file has been updated to provide more accurate type checking. Instead of simply checking if the value is an object, it now checks if it is an object with a string key and unknown value. This change improves the reliability of type checking in the codebase because in TypeScript, null is considered an object, which can lead to unexpected behavior.

Made checkResponseError more concise

### CODE REVIEW

**Review by callebtc — APPROVED** [review]

**Review by gandlafbtc — CHANGES_REQUESTED** [review]
Thanks for the PR!

Sorry i got to it so late. 

We removed axios, do you want to adapt the PR? 
I think you can remove the changes in the `checkResponseError` function. No need to change anything there, we want to revamp the error handling anyway

File: src/utils.ts
Comment (gandlafbtc) [inline src/utils.ts:48]: while we're at it, can you add the return type to this function signature?

File: src/utils.ts
Comment (gandlafbtc) [inline src/utils.ts:171]: Can you add the missing return type here?

File: src/utils.ts
Comment (gandlafbtc) [inline src/utils.ts:214]: and here

**Review by gandlafbtc — COMMENTED** [review]
Hi! Can you please add the missing return types? I think then after this we merge it. I'm still waiting on feedback on other PRs, but IDK how long this takes, so let's merge this first

**Comment by gandlafbtc** [comment]
closing due to inactivity

**Pull request closed without merging**
